src/main/java/org/folio/innreach/domain/entity/ContributionCriteriaConfiguration.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.innreach.domain.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.folio.innreach.domain.entity.base.Auditable;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+@Log4j2
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@EqualsAndHashCode(exclude = {"createdBy", "createdDate", "lastModifiedBy", "lastModifiedDate", "excludedLocations", "updateCounter", "statisticalCodeBehaviors"})
+
+@Entity
+@Table(name = "contribution_criteria_configuration")
+public class ContributionCriteriaConfiguration extends Auditable<String> {
+  @Id
+  @Column(name = "central_server_id")
+  private UUID centralServeId;
+
+  @OneToMany(
+    fetch = FetchType.LAZY,
+    cascade = CascadeType.ALL,
+    mappedBy = "contributionCriteriaConfiguration",
+    orphanRemoval = true
+  )
+  private Set<ContributionCriteriaExcludedLocation> excludedLocations = new HashSet<>();
+
+  @OneToMany(
+    fetch = FetchType.LAZY,
+    cascade = CascadeType.ALL,
+    mappedBy = "contributionCriteriaConfiguration",
+    orphanRemoval = true
+  )
+  private Set<ContributionCriteriaStatisticalCodeBehavior> statisticalCodeBehaviors = new HashSet<>();
+
+  private Integer updateCounter = 0;
+
+  public void addExcludedLocation(@NotNull ContributionCriteriaExcludedLocation excludedLocationForAdd) {
+    int hashCodeBeforeUpdate = excludedLocations.hashCode();
+    excludedLocations.add(excludedLocationForAdd);
+    if (hashCodeBeforeUpdate != excludedLocations.hashCode()) touchUpdateTrigger();
+  }
+
+  public void removeExcludedLocation(@NotNull ContributionCriteriaExcludedLocation excludedLocationForRemove) {
```

Comment:
Please, remove a reference to the parent entity to maintain the correct bidirectional relationship.
For example,

```
excludedLocations.remove(excludedLocationForRemove);
excludedLocationForRemove.setContributionCriteriaConfiguration(null);
```

Reply:
Do you mean the case when deleted entity stay in application RAM after deletion?

Reply:
No, it's not about RAM, it's about hibernate, you should always synchronize both sides of an entity relationship.
We need to have both sides in sync as otherwise, we break the Domain Model relationship consistency, and the entity state transitions are not guaranteed to work unless both sides are properly synchronized.